openwisp_controller/config/static/config/js/widget.js

```diff
@@ -153,6 +153,29 @@
         return error;
     };
 
+    var validateMaxLength = function(){
+        var config_variables = {};
+        if (window.getContext()){
+            config_variables = JSON.parse(window.getContext().value);
+        }
+        var config_variables_keys = Object.keys(config_variables);
+        var all_fields = $(".jsoneditor .vTextField");
+        all_fields.map((index, field)=>{
+            if ($(field).attr('maxLength')){
+                var maxLength = $(field).attr('maxLength');
+                $(field).on('input', () => {
+                    $(config_variables_keys).map((i,key)=>{
+                        if (key.startsWith($(field).val()) || $(field).val().startsWith('{')){
+                            $(field).removeAttr('maxLength');
+                        } else {
+                            $(field).attr('maxLength', maxLength);
+                        }
```

Comment:
How are you checking that length of the value of variable is less than maxLength set for that field?

Reply:
I have tested the same way that you have done above, but I don't know why it is not working for you.
I will test again.
I am not checking the maxLength, I am checking that if a field in config schema has max length attribute then if the value starts with `{` or the value is equal to config variable then it removes the max length attribute for that instance and when the user enters wrong value i.e. above conditions do not met then max Length attribute is again set for that field.